core/src/test/java/kafka/admin/ConfigCommandIntegrationTest.java

```diff
@@ -65,219 +75,494 @@
+                        singletonMap("ssl.keystore.location", "/tmp/test.jks")));
 
         // Per-broker config configured at default cluster-level should fail
         assertThrows(ConfigException.class,
-            () -> alterConfigWithZk(zkClient, Collections.singletonMap("listener.name.external.ssl.keystore.location", "/tmp/test.jks"), Optional.empty()));
-        deleteAndVerifyConfig(zkClient, Collections.singleton("listener.name.external.ssl.keystore.location"), Optional.of(brokerId));
+                () -> alterConfigWithZk(zkClient, adminZkClient, Optional.empty(),
+                        singletonMap("listener.name.internal.ssl.keystore.location", "/tmp/test.jks")));
+        deleteAndVerifyConfig(zkClient, adminZkClient, Optional.of(brokerId),
+                singleton("listener.name.internal.ssl.keystore.location"));
 
         // Password config update without encoder secret should fail
         assertThrows(IllegalArgumentException.class,
-            () -> alterConfigWithZk(zkClient, Collections.singletonMap("listener.name.external.ssl.keystore.password", "secret"), Optional.of(brokerId)));
+                () -> alterConfigWithZk(zkClient, adminZkClient, Optional.of(brokerId),
+                        singletonMap("listener.name.internal.ssl.keystore.password", "secret")));
 
         // Password config update with encoder secret should succeed and encoded password must be stored in ZK
         Map<String, String> configs = new HashMap<>();
-        configs.put("listener.name.external.ssl.keystore.password", "secret");
+        configs.put("listener.name.internal.ssl.keystore.password", "secret");
         configs.put("log.cleaner.threads", "2");
-        Map<String, String> encoderConfigs = Collections.singletonMap(PasswordEncoderConfigs.PASSWORD_ENCODER_SECRET_CONFIG, "encoder-secret");
-        alterConfigWithZk(zkClient, configs, Optional.of(brokerId), encoderConfigs);
+        Map<String, String> encoderConfigs = new HashMap<>(configs);
+        encoderConfigs.put(PASSWORD_ENCODER_SECRET_CONFIG, "encoder-secret");
+        alterConfigWithZk(zkClient, adminZkClient, Optional.of(brokerId), encoderConfigs);
         Properties brokerConfigs = zkClient.getEntityConfigs("brokers", brokerId);
-        assertFalse(brokerConfigs.contains(PasswordEncoderConfigs.PASSWORD_ENCODER_SECRET_CONFIG), "Encoder secret stored in ZooKeeper");
+        assertFalse(brokerConfigs.contains(PASSWORD_ENCODER_SECRET_CONFIG), "Encoder secret stored in ZooKeeper");
         assertEquals("2", brokerConfigs.getProperty("log.cleaner.threads")); // not encoded
-        String encodedPassword = brokerConfigs.getProperty("listener.name.external.ssl.keystore.password");
+        String encodedPassword = brokerConfigs.getProperty("listener.name.internal.ssl.keystore.password");
         PasswordEncoder passwordEncoder = ConfigCommand.createPasswordEncoder(JavaConverters.mapAsScalaMap(encoderConfigs));
```

Comment:
in the production code, we convert the java collection to scala for `createPasswordEncoder`. By contrast, we do a reverse conversion in testing... that is weird. Could we change the input types from scala to java?

Reply:
Sorry, I don't understand why convert Java type to Scala is weird. If I change the input type in `ConfigCommand.scala`, I should use `java.util.Map` in scala code, that isn't strange?

Reply:
sorry for unclear comment. My point was `createPasswordEncoder` should use java collection, since in production we do conversion from java to scala too. 
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/admin/ConfigCommand.scala#L248

Reply:
I understand, Thanks for your comment.